src/OrchardCore/OrchardCore/Modules/ModularBackgroundService.cs

```diff
@@ -430,4 +430,25 @@ private void CleanSchedulers(string tenant, IEnumerable<IBackgroundTask> tasks)
             }
         }
     }
+
+    private static void SetBaseUrl(HttpContext context, string baseUrl)
+    {
+        if (Uri.TryCreate(baseUrl, UriKind.Absolute, out var uri))
+        {
+            context.Request.Scheme = uri.Scheme;
+            context.Request.Host = new HostString(uri.Host, uri.Port);
+
+            // Set the PathBase only if it's not just "/". Otherwise, the original path base is kept, which may
+            // contain the RequestUrlPrefix if any.
+            if (!string.IsNullOrEmpty(uri.AbsolutePath) && uri.AbsolutePath != "/")
+            {
+                context.Request.PathBase = uri.AbsolutePath;
+            }
+
+            if (!string.IsNullOrWhiteSpace(uri.Query))
+            {
+                context.Request.QueryString = new QueryString(uri.Query);
+            }
+        }
```

Comment:
Nothing to complain about with the actual changes, but how about an early return while we're at it?


```suggestion
    private static void SetBaseUrl(HttpContext context, string baseUrl)
    {
        if (!Uri.TryCreate(baseUrl, UriKind.Absolute, out var uri))
        {
            return;
        }

        context.Request.Scheme = uri.Scheme;
        context.Request.Host = new HostString(uri.Host, uri.Port);

        // Set the PathBase only if it's not just "/". Otherwise, the original path base is kept, which may
        // contain the RequestUrlPrefix if any.
        if (!string.IsNullOrEmpty(uri.AbsolutePath) && uri.AbsolutePath != "/")
        {
            context.Request.PathBase = uri.AbsolutePath;
        }

        if (!string.IsNullOrWhiteSpace(uri.Query))
        {
            context.Request.QueryString = new QueryString(uri.Query);
        }
    }
```